pkgs/build-support/setup-hooks/reproducible-builds.sh

```diff
@@ -1,4 +1,7 @@
 # Use the last part of the out path as hash input for the build.
 # This should ensure that it is deterministic across rebuilds of the same
 # derivation and not easily collide with other builds.
-export NIX_CFLAGS_COMPILE+=" -frandom-seed=${out##*/}"
+# We also truncate the hash so that it cannot cause reference cycles.
+local outbase="${out##*/}"
+local randomseed="${outbase:0:10}"
```

Comment:
This should be good enough to make them different from previous builds of the same package. :+1:.

Reply:
I was a little concerned that the seed is the same for all files, but gcc recommends each file has a unique seed. Is it important for the seed to differ for each build too?